nestedtensor/csrc/nested_tensor_impl.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <nestedtensor/csrc/nested_tensor.h>
+
+namespace at {
+
+constexpr auto NestedTensorKey = DispatchKey::PrivateUse1_PreAutograd;
+
+struct NestedTensorImpl : public c10::TensorImpl {
+  explicit NestedTensorImpl(torch::nested_tensor::NestedTensor&& data)
+      : TensorImpl(
+            c10::DispatchKeySet(NestedTensorKey),
+            data.dtype(),
+            data.device()),
+        _data(std::move(data)) {}
+
+  int64_t dim() const override {
```

Comment:
We're assuming that these methods will stay virtual on TensorImpl. I'm not sure that that will be the case, though: for performance reasons I can see us getting rid of all virtual methods on TensorImpl, but then there should exist some sort of override mechanism for these. Might be something good to bring up with the c10 folks, it's similar to the problem of overriding `grad`